### PR TITLE
feat: mobile app refiner integration

### PR DESCRIPTION
**Pull request by Z3RO-O**
## Clickup
https://app.clickup.com/t/86cwqz04q

## Code Coverage
Please add code coverage here

## UI Preview
image

## Release Notes

- **New Features**
  - Introduced a new method to trigger NPS surveys based on user roles and feature flags in the expense management component.
  - Added tracking capabilities for user interactions through the integration of the Refiner service.

- **Bug Fixes**
  - Improved error handling and logic for checking mandatory receipts in the expense component.
  - Enhanced test coverage for various methods in the expense management component, ensuring better validation of edge cases.

- **Refactor**
  - Removed the dependency on the RefinerService from multiple components, streamlining their logic and improving maintainability.
  - Updated method signatures to enhance asynchronous handling in the report creation process.

- **Documentation**
  - Added a new environment variable for the Refiner NPS project ID to expand configuration options.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -22,6 +22,22 @@
...
+      _refiner('setProject', '786c1690-ae56-11ec-b0be-031fde0aa1f4');
+      (function () {
+        var a = document.createElement('script');
+        a.type = 'text/javascript';
+        a.async = !0;
+        a.src = 'https://js.refiner.io/v001/client.js';
+        var b = document.getElementsByTagName('script')[0];
+        b.parentNode.insertBefore(a, b);
+      })();
+    </script>
```
[the remark this answers is not in the document]
Comment (Z3RO-O) [reply src/index.html:40]: @Chethan-Fyle ,Sir how to import the id in html? I was unable to find any way to do that

[the remark this answers is not in the document]
Comment (Chethan-Fyle) [reply src/index.html:40]: Looks like we don't have `replace` in the mobile app like we do in the web app. 
Check with @suyash-patil once if initialization can be moved to the service.

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -3591,6 +3593,22 @@
...
+      .pipe(
+        switchMap(([roles, showNpsSurvey]) => {
+          if (showNpsSurvey && !roles.includes('ADMIN')) {
+            this.refinerService.startSurvey({ actionName: 'Save Expense' });
+          }
+          return of(null);
+        })
+      )
+      .subscribe();
+  }
```
[the remark this answers is not in the document]
Comment (Z3RO-O) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3610]: not required

File: src/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (Z3RO-O) [reply src/index.html]: done 👍

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/view-team-report/view-team-report.page.spec.ts [hunk 3]
```diff
@@ -566,18 +608,103 @@ describe('ViewTeamReportPageV2', () => {
...
       expect(approverReportsService.approve).toHaveBeenCalledOnceWith(platformReportData.id);
-      expect(refinerService.startSurvey).toHaveBeenCalledOnceWith({ actionName: 'Approve Report' });
       expect(router.navigate).toHaveBeenCalledOnceWith(['/', 'enterprise', 'team_reports']);
-    });
+      expect(launchDarklyService.getVariation).toHaveBeenCalledOnceWith('nps_survey', false);
+      expect(refinerService.startSurvey).toHaveBeenCalledOnceWith({ actionName: 'Approve Report' });
+    }));
 
-    it('should toggle tooltip if approval priviledge is not provided', async () => {
+    it('should toggle tooltip if approval privilege is not provided', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/view-team-report/view-team-report.page.spec.ts:616]: lol

File: src/app/fyle/view-team-report/view-team-report.page.spec.ts [hunk 4]
```diff
@@ -566,18 +608,103 @@ describe('ViewTeamReportPageV2', () => {
...
 
-      await component.approveReport();
+      component.approveReport();
+      tick();
+
       expect(component.toggleTooltip).toHaveBeenCalledTimes(1);
-    });
+    }));
+
+    it('should not approve report if user does not have approval privileges', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/view-team-report/view-team-report.page.spec.ts:627]: should we change this to `approval permissions`? privilege sounds off 🥲

**Review by Dimple16 — COMMENTED** [review]
minor comments

**Review by Dimple16 — APPROVED** [review]

**Review by arjunaj5 — APPROVED** [review]
